File: Data Structures & Algorithms/count-number-of-islands/submission-0.java
class Solution {
    public int numIslands(char[][] grid) {
        int num = 0;
        for(int i = 0; i < grid.length; i++) {
            for(int j = 0; j < grid[i].length; j++) {
                if(grid[i][j] == '1') num++;
                mark(grid, i, j);
            }
        }
        return num;
    }
    void mark(char[][] grid, int i, int j) {
        if(i < 0|| i >= grid.length || j < 0 || j >= grid[i].length) return;
        if(grid[i][j] == '0') return;
        grid[i][j] = '0';
        mark(grid, i-1, j);
        mark(grid, i+1, j);
        mark(grid, i, j-1);
        mark(grid, i, j+1);
    }
}

File: Data Structures & Algorithms/islands-and-treasure/submission-0.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        ROWS, COLS = len(grid), len(grid[0])
        directions = [(1, 0), (-1, 0), (0, 1), (0, -1)]
        INF = 2147483647

        def bfs(i,j):
            q = deque([(i, j)])
            visit = [[False] * COLS for _ in range(ROWS)]
            visit[i][j] = True
            steps = 0
            while q:
                for _ in range(len(q)):
                    row, col = q.popleft()
                    if grid[row][col] == 0:
                        return steps
                    for dr, dc in directions:
                        nr, nc = row + dr, col + dc
                        if (0 <= nr < ROWS and 0 <= nc < COLS and not visit[nr][nc] and grid[nr][nc] != -1):
                            visit[nr][nc] = True
                            q.append((nr, nc))
                steps += 1
            return INF
        
        for i in range(ROWS):
            for j in range(COLS):
                if grid[i][j] == INF:
                    grid[i][j] = bfs(i,j)

                    